src/github.ts

```diff
@@ -0,0 +1,85 @@
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import { GetResponseDataTypeFromEndpointMethod } from '@octokit/types';
+import { GitHub } from '@actions/github/lib/utils';
+import { extractBumpComment } from './common';
+
+export class Repo {
+  readonly octokit: InstanceType<typeof GitHub>;
+  readonly owner: string;
+  readonly name: string;
+  readonly prNumber?: number;
+
+  constructor() {
+    // Fetch GitHub Action context
+    // from GITHUB_REPOSITORY & GITHUB_EVENT_PATH
+    const { owner, repo, number: prNumber } = github.context.issue;
+    // const { sha, ref } = github.context;
+
+    this.owner = owner;
+    this.name = repo;
+    this.prNumber = prNumber;
+
+    this.octokit = this.getOctokit();
+  }
+
+  isPr(): boolean {
+    return !!this.prNumber;
+  }
+
+  getOctokit(): InstanceType<typeof GitHub> {
+    const ghToken = core.getInput('github-token') || process.env['GITHUB_TOKEN'];
+
+    if (!ghToken) {
+      throw new Error(
+        'No GITHUB_TOKEN env variable available. Are you sure to run this package from a Github Action?',
+      );
+    }
+
+    return github.getOctokit(ghToken);
+  }
+
+  async createOrUpdateComment(body: string, digest: string): Promise<void> {
```

Comment:
Should we extract private `findExistingComment` and `writeComment` methods so this one is bit shorter?

Reply:
I've extracted `findExistingComment` but left the “write comment” logic in the createorupdate function.